importers/data_sources/importer.py

```diff
@@ -0,0 +1,153 @@
+- uses the source_data schema to validate
+- save as separate files, one  per node
+- load and import those nodes 
+- using the same source_data schema to validate
+"""
+import argparse
+import json
+import os
+import traceback
+import requests
+import importers.utils.config as config
+from relation_engine_server.utils.json_validation import (
+    get_schema_validator,
+)
+
+
+class Importer(object):
+    def __init__(self):
+        pass
+
+    def get_config_or_fail(self, key):
+        if not hasattr(self, "_config"):
+            self._configure()
+
+        if key not in self._config:
+            raise KeyError(f'No such config key: "{key}"')
+
+        return self._config[key]
+
+    def get_config(self, key, default_value):
+        if not hasattr(self, "_config"):
+            self._configure()
+
+        if key not in self._config:
+            return default_value
+
+        return self._config[key]
+
+    def _configure(self):
+        self._config = config.load_from_env(extra_optional=["ROOT_DATA_PATH"])
+        return self._config
+
+    def get_dataset_schema_dir(self):
+        dir_path = os.path.dirname(os.path.realpath(__file__))
+        # TODO:  factor out "data_sources"
+        return os.path.join(
+            dir_path, "../", "../", "spec", "datasets", "data_sources"
+        )
+
+    def get_relative_dir(self, path):
+        dir_path = os.path.dirname(os.path.realpath(__file__))
+
+        if isinstance(path, str):
+            path = [path]
+
+        path = [dir_path] + path
+        return os.path.join(*path)
+
+    def load_data(self, dry_run=False):
+        print('[importer] Loading data')
+        print('[importer] Parameters:')
+        print(f'[importer]     API_URL: {self.get_config_or_fail("API_URL")}')
+        print(f'[importer]     dry run: {dry_run}')
+
+        # TODO: just get all files in the directory
+        default_data_dir = self.get_relative_dir('data')
+        env_data_dir = self.get_config('ROOT_DATA_PATH', None)
+        if env_data_dir is not None:
+            print('[importer]     (Taking data dir from environment variable "RES_ROOT_DATA_PATH")')
+            data_dir = env_data_dir
+        else:
+            print('[importer]     (Taking data dir from default)')
+            data_dir = default_data_dir
+        print(f'[importer]     data_dir: "{data_dir}"')
+
+        # The save_dataset method expects a list of documents
+        # to save, so we are already set!
+        schema_file = os.path.join(self.get_dataset_schema_dir(), "data_sources_nodes.yaml")
+        validator = get_schema_validator(schema_file=schema_file)
+
+        file_path = os.path.join(data_dir, 'data_sources.json')
+        with open(file_path, 'r') as data_file:
+            data_sources = json.load(data_file)
+            for data_source in data_sources:
+                if not validator.is_valid(data_source):
+                    for e in sorted(validator.iter_errors(data_source), key=str):
```

Comment:
I think a more readable version would be to remove the for loop 

```py
err = sorted(validator.iter_errors(data_source), key=str)[0]
print(f'... {err.message}')
return
```
Unless you meant to print every error, in which case the `return` needs to be unindented

Reply:
I've refactored the error handling and message printing; it looks like I didn't quite finish up the error handling and left that return in while debugging!